### PR TITLE
fasta, standard edition

### PR DESCRIPTION
**Pull request by psahabu**
@chapel-lang/perf-team:
- copied fasta.* to fastaredux.chpl
- simplified fasta.chpl to not use probability lookup scaling

### CODE REVIEW

File: test/release/examples/benchmarks/shootout/fasta.chpl [hunk 1]
```diff
@@ -111,33 +107,31 @@ proc repeatMake(desc, alu, n) {
...
     for (r, i) in zip(getRands(bytes), 0..) {
-      var ai = r: int + 1;
-      while (lookup[ai](prob) < r) do
-        ai += 1;
-      
-      line_buff[i] = lookup[ai](nucl);
+      if r < a[1](prob) {
+        line_buff[i] = a[1](nucl);
+      } else {
+        var lo = a.domain.low;
```
Comment (ben-albrecht) [inline test/release/examples/benchmarks/shootout/fasta.chpl]:
```chapel
var lo = a.domain.low,
    hi = a.domain.high;
```

or even

```chapel
var (lo, hi) = (a.domain.lo, a.domain.hi);
```

File: test/release/examples/benchmarks/shootout/fasta.chpl [hunk 2]
```diff
@@ -155,10 +149,9 @@ iter getRands(n) {
   param IA = 3877,
         IC = 29573,
         IM = 139968;
-  const SCALE = lookupScale / IM;
 
   for 0..#n {
     lastRand = (lastRand * IA + IC) % IM;
-    yield SCALE * lastRand;
+    yield lastRand:real / IM;
```
Comment (ben-albrecht) [inline test/release/examples/benchmarks/shootout/fasta.chpl]:
```chapel
yield lastRand: real / IM;
```
seems to be more consistent with spacing.

**Comment by psahabu** [comment]
@benharsh, @bradcray: care to review?

**Comment by benharsh** [comment]
Do we need to change the attribution at all?

**Comment by psahabu** [comment]
I don't think so, I didn't really do much to change this. Either you or Brad can submit it.

**Comment by bradcray** [comment]
@psahabu : Did you model the delta w.r.t. fasta-redux on any existing reference implementations?  Did you look at any existing reference implementations to validate that we'd done something reasonable/consistent with them?

This PR should include a modification to a .graph file to include this version.  I'm not sure whether to put it on the same graph as the other fasta or to have two separate graphs.  Left to my own devices, I'd put both on the same graph.

File: test/release/examples/benchmarks/shootout/fasta.chpl [hunk 3]
```diff
@@ -76,13 +73,12 @@ proc main() {
 //
 // Scan the alphabets' probabilities to compute cut-offs
 //
-proc sumAndScale(alphabet: [?D]) {
+proc sumProbs(alphabet: [?D]) {
```
Comment (bradcray) [inline test/release/examples/benchmarks/shootout/fasta.chpl]: It looks like the ': [?D]' could be dropped now that we're not referring to 'D' within this routine?

**Comment by psahabu** [comment]
I did look at the fasta by Paul Hsieh (fasta.gcc), which reminded me to do a binary search on the de-optimized probability table. Other than that I couldn't find any significant differences.

I've added the necessary credit and the .graph modification.

**Comment by bradcray** [comment]
Based on that, I'd probably update the top comment to say that it's based on the Chapel fasta-redux version and Paul Hsieh's version (using the wording we use in the other versions).  Specifically, I'd probably drop the reference to the Petr version that the Chapel fasta-redux version was based on (unless you looked at that in this effort as well).

**Comment by bradcray** [comment]
Hi Preston -- It looks like there are some problems with the graphing and that you probably didn't test it before committing:

1) the two fasta versions in the release both name 'fasta-release' in their .perfexecopts file which means that they will both be writing their data to the same .dat file (and their data will be merged into a single line as a result.  You want to give each one a unique name (say fasta-release and fastaredux-release).

2) Then you'll need to add a new entry to the .graph file to get a separate line for each of those.

3) For the new line you already added to the .graph file, it looks like you forgot to add a new 'real' entry to the top line of the .graph file.  This says "which keys from the corresponding files should be graphed?" so there should be one entry per file.  'real' is the label that's printed out when using 'time' to time things.  So you'll need to add one for the new entry you added last night and a second for the fastaredux-release vs. fasta-release entries.

I think that should do it, but once you've made those edits locally, you should run:

     start_test -performance test/release/examples/benchmarks/shootout/fasta*.chpl test/studies/shootout/fasta

and then look at the graphs that are generated by pointing a browser at test/perfdat/<yourmachinename>/html/index.html and looking at the rightmost entries on the fasta graph to see if they're all accounted for.  If working from a Mac, I believe our experience is that only Safari will render these local files properly, not Chrome.

**Comment by psahabu** [comment]
Hey Brad, sorry about that, I thought I ran the tests but clearly I don't understand the graph system very well right now. I'm working on it now, I should have it clean within the hour.

**Pull request merged**